File: frontend/src/Models/Application.ts
export interface Application {
  id: number;
  projectId: number;
  projectTitle: string;
  applicantId: string;
  applicantName: string;
  applicantEmail: string;
  message: string;
  status: ApplicationStatus;
  appliedAt: string; // ISO date string
  reviewedAt?: string; // ISO date string, optional
}

export interface ApplyRequest {
  projectId: number;
}

export interface UpdateApplicantStatusRequest {
  status: ApplicationStatus;
}

// Use union type instead of enum for erasableSyntaxOnly compatibility
export type ApplicationStatus = 'Pending' | 'Approved' | 'Denied' | 'Withdrawn';

File: backend/Models/Application.cs
namespace Models
{
    public class Application
    {
        public long Id { get; set; }
        
        // Foreign key to Project
        public int ProjectId { get; set; }
        
        // Foreign key to User (applicant)
        public string ApplicantId { get; set; } = null!;
        
        // Application status
        public ApplicationStatus Status { get; set; } = ApplicationStatus.Pending;
        
        // When the application was submitted
        public DateTime AppliedAt { get; set; } = DateTime.UtcNow;
        
        // When the application was reviewed (approved/denied)
        public DateTime? ReviewedAt { get; set; }
        
        // Navigation properties
        public Project Project { get; set; } = null!;
        public User Applicant { get; set; } = null!;
    }
    
    public enum ApplicationStatus
    {
        Pending = 0,
        Approved = 1,
        Denied = 2,
        Withdrawn = 3
    }
}
